internal/interop/interopconnect/test_cases.go

```diff
@@ -298,23 +299,38 @@ func validateMetadata(
 	expectedBinaryHeaders map[string][][]byte,
 ) {
 	for key, values := range expectedStringHeaders {
-		assert.Equal(t, len(values), len(header.Values(key)))
+		actualValues := header.Values(key)
+		// If the returned header values are not equal to what we expect, next check whether the header
+		// values were combined into one comma-delimited string. The Fetch API Headers object will return
+		// duplicate headers this way.
+		if len(values) != len(actualValues) && len(actualValues) == 1 {
+			actualValues = strings.Split(actualValues[0], ", ")
+		}
```

Comment:
We're not covering all cases here. All following forms are semantically equivalent:
```
Foo: a
Foo: b
Foo: c
Foo: d
```

```
Foo: a, b, c, d
```

```
Foo: a
Foo: b, c, d
```

We need a comment that this is not a complete semantic comparison.

Reply:
Would that third case ever occur though? I guess if a non-Fetch API server decided to concatenate values on their own and add them as a duplicate?

I added a comment too.  Lmk what you think.

Reply:
I don't know if that will ever occur. It would be best if validateMetadata() did a semantical comparison, so that this uncertainty is removed. Doing so will end up as a significant piece of code that needs test coverage on its own - so the status quo with a comment is acceptable in my book.